mars/services/web/ui/src/task_info/TaskTileableGraph.js

```diff
@@ -58,10 +68,88 @@ export default class TaskTileableGraph extends React.Component {
 
     /* eslint no-unused-vars: ["error", { "args": "none" }] */
     componentDidUpdate(prevProps, prevStates, snapshot) {
-        if (prevStates.tileables !== this.state.tileables
-                && prevStates.dependencies !== this.state.dependencies) {
+        if (Object.keys(this.state.tileableDetails).length !== this.state.tileables.length) {
+            return;
+        }
+
+        if ((prevStates.tileables !== this.state.tileables
+                && prevStates.dependencies !== this.state.dependencies)
+                || prevStates.tileableDetails !== this.state.tileableDetails) {
             d3Select('#svg-canvas').selectAll('*').remove();
 
+            // Set up an SVG group so that we can translate the final graph.
+            const svg = d3Select('#svg-canvas'),
+                inner = svg.append('g');
+
+            svg
+                .append("circle")
```

Comment:
Make a utility function to create these shapes? Code a little redundant for me. Comments also needed to show what is actually done here.